src/c_wrapper.jl

```diff
@@ -29,14 +29,32 @@ function SCS_solve(T::Union{Type{Direct}, Type{Indirect}},
         m::Int, n::Int, A::SCSVecOrMatOrSparse, b::Array{Float64},
         c::Array{Float64}, f::Int, l::Int, q::Array{Int}, s::Array{Int},
         ep::Int, ed::Int, p::Array{Float64},
-        primal_sol::Vector{Float64}=Float64[],
-        dual_sol::Vector{Float64}=Float64[],
-        slack::Vector{Float64}=Float64[];
+        primal_sol::Vector{Float64}=zeros(n),
+        dual_sol::Vector{Float64}=zeros(m),
+        slack::Vector{Float64}=zeros(m);
         options...)
 
     n > 0 || throw(ArgumentError("The number of variables in SCSModel must be greater than 0"))
     m > 0 || throw(ArgumentError("The number of constraints in SCSModel must be greater than 0"))
 
+    ws = (:warm_start=>true) in options
+    warmstart_sizes_are_correct = length(primal_sol) == n && length(dual_sol) == length(slack) == m
+
+    if warmstart_sizes_are_correct
+        if !ws
+            primal_sol = fill!(primal_sol, 0.0)
```

Comment:
Continuing the discussion from the last PR (https://github.com/JuliaOpt/SCS.jl/pull/164/files#r360657916):

> that's true, but only since these are Vectors (we require this in the signature) and these are mutable. Is it guaranteed by any means?

`fill!` is guaranteed to mutate its first argument by definition.

> I thought the best practice was to use only the return value of the mutating call. But sure, I can change this if You wish.

Where is this practice discussed? I haven't seen it before, and it doesn't make much sense to me. What if a function mutates multiple arguments?

Reply:
This is just a side note:
I thought that this is the standard practice, e.g. with mutable arithmetic types when writing generic, fast (allocationless) code which works for both mutable and immutable types: e.g. you need to write `z = mul!(z, x, y)` since just `mul!(z, x, y)` while correct for mutables, will discard the value when `z` can not be mutated, i.e. mutability is an implementation detail.

Mutability of `Vectors` will not change in foreseeable future, I completely agree ;)